hls-plugin-api/src/Ide/Types.hs

```diff
@@ -877,6 +886,57 @@ type CommandFunction ideState a
 
 -- ---------------------------------------------------------------------
 
+type ResolveFunction ideState a (m :: Method ClientToServer Request) =
```

Comment:
These should probably go in `Resolve.hs`?

Reply:
I left all the core Resolve stuff in Types, and moved the optional handler makers to Resolve. I am happy to move everything to resolve thought if you think that's better.